src/uct/ib/base/ib_iface.c

```diff
@@ -548,8 +549,25 @@ ucs_status_t uct_ib_iface_get_device_address(uct_iface_h tl_iface,
     return UCS_OK;
 }
 
+static int uct_ib_iface_roce_is_reachable_via_routing(unsigned subnet_mask,
+        const uct_ib_device_gid_info_t *local_gid_info,
+        const uct_ib_address_t *remote_ib_addr)
+{
+    size_t size;
+
+    if (!uct_ib_device_is_potentially_reachable(subnet_mask, local_gid_info)) {
+        return 0;
+    }
+
+    ucs_address_family_sizeof_ip(local_gid_info->roce_info.addr_family, &size);
+    size = sizeof(local_gid_info->gid) + (subnet_mask >> 3) - size;
```

Comment:
why >> 3?

Reply:
transfer from bits to bytes (see comment on missing bitwise support I'll add if the general approach is ok)

Reply:
ok